File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportC#/Model/Entity.cs
namespace FirmaTransportC_.Model
{
    public abstract class Entity<TId>
    {
        private TId id;

        public Entity(TId id)
        {
            this.id = id;
        }

        public TId Id
        {
            get { return id; }
            set { id = value; }
        }
    }
}

File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportJava/Model/src/main/java/transport/model/Entity.java
package transport.model;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

import java.io.Serializable;

@MappedSuperclass
public abstract class Entity<ID> implements Serializable {
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private ID id;

    protected Entity() {}

    public Entity(ID id) {
        this.id = id;
    }

    public ID getId() {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }
}
